pkgs/stdenv/generic/make-derivation.nix

```diff
@@ -177,14 +177,14 @@ rec {
            "checkInputs" "installCheckInputs"
            "__impureHostDeps" "__propagatedImpureHostDeps"
            "sandboxProfile" "propagatedSandboxProfile"])
-        // (lib.optionalAttrs (name == "")) {
+        // (lib.optionalAttrs (!(attrs ? name))) {
```

Comment:
We probably should also do ```&& attrs ? pname && attrs ? version``` to let Nix show it's default message if name is missing:

https://github.com/NixOS/nix/blob/6024dc1d97212130c19d3ff5ce6b1d102837eee6/src/libexpr/primops.cc#L530